native/index.d.ts

```diff
@@ -0,0 +1,68 @@
+// <reference path="../react/react-native.d.ts" />
+
+declare module "styled-components/native" {
```

Comment:
If you don't do this, you're going to cause problems if two different TypeScript libraries ever depend on `styled-components` and are consumed. For instance:

```
styled-components
     /   \
     A    B
     \   /
       C
```